src/provider/mod.rs

```diff
@@ -654,13 +661,13 @@ impl RpcHandler {
     async fn id(self, _: IdRequest) -> IdResponse {
         IdResponse {
             peer_id: Box::new(self.inner.keypair.public().into()),
-            listen_addrs: self.inner.listen_addresses().await.unwrap_or_default(),
+            listen_addrs: self.inner.listen_addresses().unwrap_or_default(),
             version: env!("CARGO_PKG_VERSION").to_string(),
         }
     }
     async fn addrs(self, _: AddrsRequest) -> AddrsResponse {
```

Comment:
this does not have to be async anymore, right?

Reply:
yes, but the rpc code was upset when I made it not async 🤷